api_tests/lib_sdk/wallets/lightning.ts

```diff
@@ -0,0 +1,102 @@
+import { pollUntilMinted, Wallet } from "./index";
+import { Asset } from "../asset";
+import BigNumber from "bignumber.js";
+import BitcoinRpcClient from "bitcoin-core";
+import { BitcoinNodeConfig } from "../../lib/bitcoin";
+import { toBitcoin } from "satoshi-bitcoin";
+import { Lnd } from "../lnd";
+import { HarnessGlobal } from "../../lib/util";
+import { Logger } from "log4js";
+import { E2ETestActorConfig } from "../../lib/config";
+
+declare var global: HarnessGlobal;
+
+export class LightningWallet implements Wallet {
+    public static async newInstance(
+        bitcoinNodeConfig: BitcoinNodeConfig,
+        logger: Logger,
+        logDir: string,
+        actorConfig: E2ETestActorConfig
+    ) {
+        const bitcoinRpcClient = new BitcoinRpcClient({
```

Comment:
Can we reuse `bitcoinWallet.ts` for this? They should be connected to the same chain right? 

you could then define LightningWallet like this: 

```
public static async newInstance(
        bitcoinNodeConfig: BitcoinNodeConfig,
        logger: Logger,
        logDir: string,
        actorConfig: E2ETestActorConfig
    ) {
        const bitcoinWallet = BitcoinWallet.newInstance({...
```

and then further down you don't have to reimplement the logic for minting but can refer to:

```
bitcoinWallet.mint(...)
```

or different idea: Do you need this function in here? Can't you use: 

```
Actor.fund(ln.createChainAddress())
```

My assumption is no, because `mint` is somewhere called automatically based on the interface `Wallet`.

Reply:
`mint` is part of the wallet interface yes.

I am in favor of this proposal, reusing the bitcoin wallet sounds like a good idea to reduce the code duplication.